pkg/monitor/cluster/debugpods_test.go

```diff
@@ -0,0 +1,83 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	ktesting "k8s.io/client-go/testing"
+
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitDebugPodsCount(t *testing.T) {
+	ctx := context.Background()
+
+	cli := fake.NewSimpleClientset(
+		&corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "master-1",
+			},
+		},
+		&corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "master-2",
+			},
+		},
+	)
+
+	now := time.Now()
+	longAgo := time.Date(1991, time.August, 24, 0, 0, 0, 0, time.UTC)
+	reactorFn := func(action ktesting.Action) (handled bool, ret kruntime.Object, err error) {
+		return true, &corev1.EventList{Items: []corev1.Event{
```

Comment:
I think there should also be a test that tests that pod started events for non debug pods are not counted

Reply:
It is not yet possible to test k8s API call settings, including field-selector
https://github.com/kubernetes/client-go/issues/326